.github/workflows/end2end.yaml

```diff
@@ -742,6 +742,37 @@ jobs:
           source: /tmp/artifacts
         if: always()
 
+  mongo-patches-check:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: run patches
+        run: make -C solution-base/mongodb patch
+      - name: Check if patches were applied right
+        run: |
+          git diff --exit-code || { echo "Error: Changes detected after running 'make fetch-mongodb-sharded'. Please ensure step is completed."; exit 1; }
+      - name: Set status in ARTIFACTS to enable re-run only failed jobs
+        if: success()
+        run: |
+          mkdir -p /tmp/artifacts/build_status/${{ github.job }}
+          echo 'SUCCESSFUL' > /tmp/artifacts/build_status/${{ github.job }}/.final_status
+      - name: Set status in ARTIFACTS to enable re-run only failed jobs
+        if: failure()
+        run: |
+          mkdir -p /tmp/artifacts/build_status/${{ github.job }}
+          echo 'FAILED' > /tmp/artifacts/build_status/${{ github.job }}/.final_status
```

Comment:
Is this really needed? We already have `write-final-success-status` and `write-final-failed-status` to handle that at the end of the build, if anything failed